crates/rnote-cli/src/export.rs

```diff
@@ -0,0 +1,673 @@
+        ext => Err(anyhow::anyhow!(
+            "Could not create doc export prefs, unsupported export file extension `{ext}`"
+        )),
+    }
+}
+pub(crate) fn create_selection_export_prefs_from_args(
+    output_file: Option<impl AsRef<Path>>,
+    output_format: Option<SelectionExportFormat>,
+    background: bool,
+    pattern: bool,
+    bitmap_scalefactor: f64,
+    jpeg_quality: u8,
+    margin: f64,
+) -> anyhow::Result<SelectionExportPrefs> {
+    let format = match (output_file, output_format) {
+        (Some(file), None) => match file.as_ref().extension().and_then(|ext| ext.to_str()) {
+            Some(extension) => get_selection_export_format(extension),
+            None => {
+                return Err(anyhow::anyhow!(
+                    "Output file needs to have an extension to determine the file type"
+                ))
+            }
+        },
+        (None, Some(out_format)) => Ok(out_format),
+        // unreachable because they are exclusive (conflicts_with)
+        (Some(_), Some(_)) => {
+            return Err(anyhow::anyhow!(
+                "--output-file and --output-format are mutually exclusive."
+            ))
+        }
+        // unreachable because they are required
+        (None, None) => {
+            return Err(anyhow::anyhow!(
+                "--output-file or --output-format is required."
+            ))
+        }
+    }?;
+
+    let prefs = SelectionExportPrefs {
+        export_format: format,
+        with_background: background,
+        with_pattern: pattern,
+        bitmap_scalefactor,
+        jpeg_quality,
+        margin,
+    };
+
+    Ok(prefs)
+}
+
+fn get_selection_export_format(format: &str) -> anyhow::Result<SelectionExportFormat> {
+    match format {
+        "svg" => Ok(SelectionExportFormat::Svg),
+        "png" => Ok(SelectionExportFormat::Png),
+        "jpg" | "jpeg" => Ok(SelectionExportFormat::Jpeg),
+        ext => Err(anyhow::anyhow!(
+            "Could not create selection export prefs, unsupported export file extension `{ext}`"
+        )),
+    }
+}
+
+pub(crate) fn check_file_conflict(
+    output_file: &Path,
+    mut on_conflict: OnConflict,
+    on_conflict_overwrite: &mut Option<OnConflict>,
+) -> anyhow::Result<Option<PathBuf>> {
+    if !output_file.exists() {
+        return Ok(None);
+    }
+    match on_conflict_overwrite {
+        Some(o) => on_conflict = *o,
+        None => {
+            let options = &[
+                OnConflict::Ask,
+                OnConflict::Overwrite,
+                OnConflict::AlwaysOverwrite,
+                OnConflict::Skip,
+                OnConflict::AlwaysSkip,
+                OnConflict::Suffix,
+                OnConflict::AlwaysSuffix,
+            ];
+            while matches!(on_conflict, OnConflict::Ask) {
+                match dialoguer::Select::new()
+                    .with_prompt(format!("File {} already exits:", output_file.display()))
+                    .items(options)
```

Comment:
Is it possible to make the first option the default selected? Because right now it is not entirely clear that this is a prompt when it appears, and a initial `>` might make it more clear that the user is able to interact with it.

Reply:
I made it default to overwrite since it's the option that will likely be chosen the most